tooling/miniutil/src/build/terminator.rs

```diff
@@ -1,5 +1,243 @@
 use crate::build::*;
 
+impl FunctionBuilder {
+    fn finish_block(&mut self) {
+        let Some(terminator) = self.terminator else {
+            panic!("Cannot finish block as block has no terminator")
+        };
+        let bb = BasicBlock { statements: self.statements, terminator };
+        self.blocks.try_insert(self.block_name, bb).unwrap();
+    }
+
+    fn new_block(&mut self, name: u32) {
+        self.block_name = BbName(Name::from_internal(name));
+        self.statements = Default::default();
+        self.terminator = None;
+    }
+
+    /// create new block without branching
+    fn prepare_next_block(&mut self) {
+        self.new_block(self.next_block);
+        self.next_block += 1;
+    }
+
+    // terminators with 0 following blocks
+    pub fn exit(&mut self) {
+        self.terminator = Some(exit());
+        self.finish_block();
+    }
```

Comment:
It would be good if something like this led to an error:
```rust
f.exit();
f.exit();
```
So, `finish_block` should leave the builder in a state where there is *no* current block, and then any attempt at adding a statement or terminator should fail.

In `FunctionBuilder`, this could look like
```rust
cur_block: Option<CurBlock>,
```
with
```rust
struct CurBlock {
    statements: List<Statement>,
    block_name: BbName,
}
```
`CurBlock` has no terminator as adding the terminator finishes the block and makes it not the current block any more.

You probably want a `fn cur_block(&mut self) -> &mut CurBlock` to easily get access to the current block with a nice panic when there is no such block.